transaction-status/src/lib.rs

```diff
@@ -397,6 +399,51 @@ pub struct ConfirmedBlock {
     pub block_height: Option<u64>,
 }
 
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct VersionedConfirmedBlock {
+    pub previous_blockhash: String,
+    pub blockhash: String,
+    pub parent_slot: Slot,
+    pub transactions: Vec<VersionedTransactionWithStatusMeta>,
+    pub rewards: Rewards,
+    pub block_time: Option<UnixTimestamp>,
+    pub block_height: Option<u64>,
+}
+
+impl VersionedConfirmedBlock {
+    /// Downgrades a versioned block into a legacy block type
+    /// if it only contains legacy transactions
+    pub fn into_legacy_block(self) -> Option<ConfirmedBlock> {
```

Comment:
nit: any reason for all of the `into_legacy_*` functions to not implement this as `TryFrom` instead?

Reply:
No big reason, just felt this naming was better self-documentation